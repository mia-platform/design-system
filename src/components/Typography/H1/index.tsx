import { Typography as AntTypography } from 'antd'
import { ReactElement } from 'react'

import { TypographyProps } from '..'

const { Title: AntTitle } = AntTypography

/**
 * UI component to display H1 headers
 *
 * @link https://ant.design/components/typography
 * @returns {H1} H1 component
 */
export const H1 = ({
  children,
}: TypographyProps): ReactElement => {
  return (
    <AntTitle ellipsis level={1} >
      {children}
    </AntTitle>
  )
}
