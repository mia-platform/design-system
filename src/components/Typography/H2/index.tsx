import { Typography as AntTypography } from 'antd'
import { ReactElement } from 'react'

import { TypographyProps } from '..'

const { Title: AntTitle } = AntTypography

/**
 * UI component to display H2 headers
 *
 * @link https://ant.design/components/typography
 * @returns {H2} H2 component
 */
export const H2 = ({
  children,
}: TypographyProps): ReactElement => {
  return (
    <AntTitle ellipsis level={2} >
      {children}
    </AntTitle>
  )
}
