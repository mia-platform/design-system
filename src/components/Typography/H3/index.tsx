import { Typography as AntTypography } from 'antd'
import { ReactElement } from 'react'

import { TypographyProps } from '..'

const { Title: AntTitle } = AntTypography

/**
 * UI component to display H3 headers
 *
 * @link https://ant.design/components/typography
 * @returns {H3} H3 component
 */
export const H3 = ({
  children,
}: TypographyProps): ReactElement => {
  return (
    <AntTitle ellipsis level={3} >
      {children}
    </AntTitle>
  )
}
