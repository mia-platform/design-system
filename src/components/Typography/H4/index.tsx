import { Typography as AntTypography } from 'antd'
import { ReactElement } from 'react'

import { TypographyProps } from '..'

const { Title: AntTitle } = AntTypography

/**
 * UI component to display H4 headers
 *
 * @link https://ant.design/components/typography
 * @returns {H4} H4 component
 */
export const H4 = ({
  children,
}: TypographyProps): ReactElement => {
  return (
    <AntTitle ellipsis level={4} >
      {children}
    </AntTitle>
  )
}
