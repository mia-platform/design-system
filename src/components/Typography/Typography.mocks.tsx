/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Icon } from '../Icon'

export const fontUrl = 'https://fontsource.org/fonts/inter'

export const customEllipsis = {
  rows: 1,
  suffix: ' END',
  expandable: true,
  symbol: 'Show more',
}

export const customCopyable = {
  text: 'Custom text to copy',
  icon: [
    <Icon
      color="magenta"
      key="FiCopy"
      name="FiCopy"
      size={16}
    />,
    <Icon
      color="magenta"
      key="FiCheck"
      name="FiCheck"
      size={16}
    />,
  ],
  tooltips: ['Copy to clipboard!', 'Copied to clipboard!'],
}

/* Create an ad-hoc utils file in case it is used by components outside Typography */
export const loremIpsum = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex.`
