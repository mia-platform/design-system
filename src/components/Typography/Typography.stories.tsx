import type { Meta, StoryObj } from '@storybook/react'

import { Typography } from '.'

const meta = {
  title: 'Typography',
  args: {
    children: 'Text',
  },
  argTypes: {
    children: { control: { type: 'text' } },
  },
} satisfies Meta

export default meta
type Story = StoryObj<typeof meta>

export const All: Story = {
  argTypes: {
    children: { control: 'hidden' },
  },
  render: () => (
    <>
      <Typography.H1>{'H1'}</Typography.H1>
      <Typography.H2>{'H2'}</Typography.H2>
      <Typography.H3>{'H3'}</Typography.H3>
      <Typography.H4>{'H4'}</Typography.H4>
    </>
  ),
}

export const H1: Story = {
  render: (args) => <Typography.H1 {...args} />,
}

export const H2: Story = {
  render: (args) => <Typography.H2 {...args} />,
}

export const H3: Story = {
  render: (args) => <Typography.H3 {...args} />,
}

export const H4: Story = {
  render: (args) => <Typography.H4 {...args} />,
}
