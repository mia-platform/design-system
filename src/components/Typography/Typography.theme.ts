/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ComponentsTheme } from '../ThemeProvider/Ant'
import Theme from '../../themes/schema'

/**
 * Computes the relative line height of a TextFormat element from its absolute line height and font size.
 *
 * @param {number} lineHeight - Absolute line height of the TextFormat element.
 * @param {number} fontSize - Absolute font size of the TextFormat element.
 * @returns {number | undefined} relative line height (or undefined if one of the absolute values is missing).
 */
const getRelativeLineHeight = (lineHeight?: number, fontSize?: number): number | undefined => {
  if (!lineHeight || !fontSize) { return undefined }

  return lineHeight / fontSize
}

/**
 * Generates a Ant theme configuration for Typography component based on a theme configuration.
 *
 * @link https://ant.design/components/typography#design-token
 *
 * @param {Partial<Theme>} theme - theme configuration.
 * @returns {Partial<ComponentsTheme>} The generated Typography Ant theme configuration.
 */
export default ({ typography }: Partial<Theme>): ComponentsTheme['Typography'] => ({
  fontSizeHeading1: typography?.h1?.fontSize,
  fontSizeHeading2: typography?.h2?.fontSize,
  fontSizeHeading3: typography?.h3?.fontSize,
  fontSizeHeading4: typography?.h4?.fontSize,
  lineHeightHeading1: getRelativeLineHeight(typography?.h1?.lineHeight, typography?.h1?.fontSize),
  lineHeightHeading2: getRelativeLineHeight(typography?.h2?.lineHeight, typography?.h2?.fontSize),
  lineHeightHeading3: getRelativeLineHeight(typography?.h3?.lineHeight, typography?.h3?.fontSize),
  lineHeightHeading4: getRelativeLineHeight(typography?.h4?.lineHeight, typography?.h4?.fontSize),
})
