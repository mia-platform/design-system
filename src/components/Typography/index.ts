/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { H1 } from './HX/H1'
import { H2 } from './HX/H2'
import { H3 } from './HX/H3'
import { H4 } from './HX/H4'

export const Typography = {
  H1,
  H2,
  H3,
  H4,
}
