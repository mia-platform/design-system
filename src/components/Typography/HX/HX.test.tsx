/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { customCopyable, customEllipsis, loremIpsum } from '../Typography.mocks'
import { render, screen } from '../../../test-utils'
import { Typography } from '..'

const { H1, H2, H3, H4 } = Typography

describe('HX Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders H1 correctly', () => {
    const { asFragment } = render(<H1>{'Text'}</H1>)

    const heading = screen.getByRole('h1', { name: 'Text' })
    expect(heading).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders H2 correctly', () => {
    const { asFragment } = render(<H2>{'Text'}</H2>)

    const heading = screen.getByRole('h2', { name: 'Text' })
    expect(heading).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders H3 correctly', () => {
    const { asFragment } = render(<H3>{'Text'}</H3>)

    const heading = screen.getByRole('h3', { name: 'Text' })
    expect(heading).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders H4 correctly', () => {
    const { asFragment } = render(<H4>{'Text'}</H4>)

    const heading = screen.getByRole('h4', { name: 'Text' })
    expect(heading).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders custom ellipsis correctly', () => {
    const { asFragment } = render(<H1 ellipsis={customEllipsis}>{loremIpsum}</H1>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders copyable correctly', () => {
    const { asFragment } = render(<H1 copyable>{loremIpsum}</H1>)

    const copyButton = screen.getByRole('button', { name: 'Copy' })
    const copyImg = screen.getByRole('img', { name: 'copy' })
    expect(copyButton).toBeVisible()
    expect(copyImg).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders custom copyable correctly', () => {
    const { asFragment } = render(<H1 copyable={customCopyable}>{loremIpsum}</H1>)

    const copyButton = screen.getByRole('button', { name: 'Copy to clipboard!' })
    const copyImg = screen.getByRole('img', { name: 'FiCopy' })
    expect(copyButton).toBeVisible()
    expect(copyImg).toBeVisible()

    expect(asFragment()).toMatchSnapshot()
  })
})
