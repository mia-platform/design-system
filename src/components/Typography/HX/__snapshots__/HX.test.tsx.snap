// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HX Component renders H1 correctly 1`] = `
<DocumentFragment>
  <h1
    aria-label="Text"
    class="ant-typography"
    role="h1"
  >
    Text
  </h1>
</DocumentFragment>
`;

exports[`HX Component renders H2 correctly 1`] = `
<DocumentFragment>
  <h2
    aria-label="Text"
    class="ant-typography"
    role="h2"
  >
    Text
  </h2>
</DocumentFragment>
`;

exports[`HX Component renders H3 correctly 1`] = `
<DocumentFragment>
  <h3
    aria-label="Text"
    class="ant-typography"
    role="h3"
  >
    Text
  </h3>
</DocumentFragment>
`;

exports[`HX Component renders H4 correctly 1`] = `
<DocumentFragment>
  <h4
    aria-label="Text"
    class="ant-typography"
    role="h4"
  >
    Text
  </h4>
</DocumentFragment>
`;

exports[`HX Component renders copyable correctly 1`] = `
<DocumentFragment>
  <h1
    aria-label="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex."
    class="ant-typography"
    role="h1"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex.
    <div
      aria-label="Copy"
      class="ant-typography-copy"
      role="button"
      style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
      tabindex="0"
    >
      <span
        aria-label="copy"
        class="anticon anticon-copy"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="copy"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
          />
        </svg>
      </span>
    </div>
  </h1>
</DocumentFragment>
`;

exports[`HX Component renders custom copyable correctly 1`] = `
<DocumentFragment>
  <h1
    aria-label="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex."
    class="ant-typography"
    role="h1"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex.
    <div
      aria-label="Copy to clipboard!"
      class="ant-typography-copy"
      role="button"
      style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
      tabindex="0"
    >
      <svg
        alt="FiCopy"
        aria-label="FiCopy"
        color="magenta"
        fill="none"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: magenta;"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="13"
          rx="2"
          ry="2"
          width="13"
          x="9"
          y="9"
        />
        <path
          d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
        />
      </svg>
    </div>
  </h1>
</DocumentFragment>
`;

exports[`HX Component renders custom ellipsis correctly 1`] = `
<DocumentFragment>
  <h1
    aria-label="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex."
    class="ant-typography ant-typography-ellipsis ant-typography-single-line"
    role="h1"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel dolor ac sem congue ornare rhoncus quis tortor.
Vivamus hendrerit sed nisl eget gravida. Nunc dolor elit, condimentum non ante vitae, ullamcorper dignissim dolor.
Quisque ornare interdum massa blandit accumsan. Morbi viverra, urna ut consequat maximus, risus eros molestie augue, at posuere lacus nunc id ex.
Aenean et ante augue. Donec vitae pretium tellus, mattis placerat nisl. Phasellus dolor urna, porttitor a tempus at, scelerisque id nisi.
In ac libero quis neque tempor iaculis sit amet non sapien. Aenean quis neque varius, efficitur erat et, faucibus tortor.
Etiam at mi lacinia, gravida diam ac, sodales elit. Maecenas vitae rutrum nisl.
Nunc pellentesque, arcu vitae bibendum rutrum, mi sapien aliquam ipsum, sit amet mollis quam orci quis est.
Curabitur sagittis mauris volutpat lorem elementum, nec porttitor eros commodo. Suspendisse lorem orci, commodo mollis quam eget, facilisis tincidunt mauris.
Ut in feugiat nunc, in eleifend ex. END
    <span
      aria-hidden="true"
      style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; font-size: 0px; word-break: keep-all; white-space: nowrap;"
    >
      lg
    </span>
    <span
      aria-hidden="true"
      style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; font-size: 0px; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
       END
      <a
        aria-label="Expand"
        class="ant-typography-expand"
      >
        Show more
      </a>
    </span>
  </h1>
</DocumentFragment>
`;
